engine/src/main/java/org/hibernate/search/engine/search/aggregation/dsl/impl/AvgAggregationFieldStepImpl.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.search.aggregation.dsl.impl;
+
+import org.hibernate.search.engine.search.aggregation.dsl.AvgAggregationFieldStep;
+import org.hibernate.search.engine.search.aggregation.dsl.AvgAggregationOptionsStep;
+import org.hibernate.search.engine.search.aggregation.dsl.spi.SearchAggregationDslContext;
+import org.hibernate.search.engine.search.aggregation.spi.AggregationTypeKeys;
+import org.hibernate.search.engine.search.aggregation.spi.MetricDoubleAggregationBuilder;
+import org.hibernate.search.engine.search.predicate.dsl.SearchPredicateFactory;
+
+public class AvgAggregationFieldStepImpl<PDF extends SearchPredicateFactory>
+		implements AvgAggregationFieldStep<PDF> {
+	private final SearchAggregationDslContext<?, ? extends PDF> dslContext;
+
+	public AvgAggregationFieldStepImpl(SearchAggregationDslContext<?, ? extends PDF> dslContext) {
+		this.dslContext = dslContext;
+	}
+
+	@Override
+	public AvgAggregationOptionsStep<?, PDF> field(String fieldPath) {
+		MetricDoubleAggregationBuilder builder = dslContext.scope()
+				.fieldQueryElement( fieldPath, AggregationTypeKeys.AVG ).type();
```

Comment:
a double average may not be the best for the temporal types. any chance we can follow the same pattern as for the sum?